File: hw2/hello.php
<!DOCTYPE HTML>
<html lang="en-US">
<head>
</head>

<body>
	
<?php
	echo '<p>Hello World from PHP on ' . date('Y-m-d h:i:s a') . '</p>';


	$array = array("red", "white", "blue");



	$k = array_rand($array);
	$v = $array[$k];
// 	function random_color_part() {
//     return str_pad( dechex( mt_rand( 0, 255 ) ), 2, '0', STR_PAD_LEFT);
// }

// 	function random_color() {
//     return random_color_part() . random_color_part() . random_color_part();
// }

// $rand_color = random_color();



	$rand_color = $v;
echo "<body style='background-color:$rand_color'>";


?>

</body>

</html>

File: hw2/hello.pl
#!/usr/bin/perl
use CGI qw/:standard :html3/;
# print "Content-type: text/html\n\n";
# print "Hello world from CGI the time is \n";
# $now_string = localtime();
# print "$now_string";

# my @set = ('0' ..'9', 'A' .. 'F');
# my $random_color = join '' => map $set[rand @set], 1 .. 6;


my @colors = ("red","white","blue");
my $random_color = $colors[rand @colors];

# # # print $random_color;

# print body();



$newStyle=<<END;
<!-- 

Body {
	background-color: $random_color
}
P.Tip {
margin-right: 50pt;
margin-left: 50pt;
    color: red;
}
P.Alert {
font-size: 30pt;
    font-family: sans-serif;
  color: red;
}
-->
END
print header();
print start_html( 
	-title=>'Hello CGI',
	-style=>
	{
	    -code=>$newStyle}
);

print "Hello world from CGI the time is \n";
$now_string = localtime();
print "$now_string";


# print h1('CGI with Style'),
#       p({-class=>'Tip'},
#     "Better read the cascading style sheet spec before playing with this!"),
#       span({-style=>'color: magenta'},
#        "Look Mom, no hands!",
#        p(),
#        "Whooo wee!"
#        );
print end_html;